### DISCUSSION TITLE
Is it possible to use AutoUpdater.NET or some parts of it on an ASP.NET API

### DISCUSSION THREAD

**Discussion by MidKnightXI** [ravibpatel/AutoUpdater.NET#640] [Q&A] [answered]
Hello, I'm currently facing a problem with an API I'm building in ASP.NET, I need to integrate an auto-updater to make it able to update automagically on our clients machine.

I looked at some issues on StackOverflow and everything but nothing is satisfying me atm and I found this project: AutoUpdater.NET.

I know that there are some dependencies to WPF/WinForms but I'd like to know if some parts might be usable in an API in ASP.NET

If I'm unclear on what I said or what I want I'd be glad to rephrase my needs!

Thank you for helping :)

**Comment by ravibpatel** [comment, marked as answer]
You won't be able to use it directly, but there are parts of code you can easily use in your ASP.NET project. For example, update checking logic as shown [here](https://github.com/ravibpatel/AutoUpdater.NET/blob/master/AutoUpdater.NET/AutoUpdater.cs#L388) should work without much modification. Also, you can use download logic from DownloadUpdateDialog.cs too. Instead of using a separate executable like ZipExtractor you can use the code of ZipExtractor directly after the download finishes.
